beacon-chain/forkchoice/doubly-linked-tree/optimistic_sync.go

```diff
@@ -4,20 +4,51 @@ import (
 	"context"
 )
 
+func (s *Store) setOptimisticToInvalid(ctx context.Context, root, payload [32]byte) ([][32]byte, error) {
+	s.nodesLock.Lock()
+	invalidRoots := make([][32]byte, 0)
+	node, ok := s.nodeByRoot[root]
+	if !ok || node == nil {
+		s.nodesLock.Unlock()
+		return invalidRoots, ErrNilNode
+	}
+	// Check if last valid hash is an ancestor of the passed node.
+	lastValid, ok := s.nodeByPayload[payload]
+	if !ok || lastValid == nil {
+		s.nodesLock.Unlock()
+		return invalidRoots, errInvalidOptimisticStatus
+	}
+	firstInvalid := node
+	for ; firstInvalid.parent != nil && firstInvalid.parent.payloadHash != payload; firstInvalid = firstInvalid.parent {
+		if ctx.Err() != nil {
+			s.nodesLock.Unlock()
+			return invalidRoots, ctx.Err()
+		}
+	}
+	if firstInvalid.parent == nil {
+		firstInvalid = node
+	}
+	s.nodesLock.Unlock()
+	if err := lastValid.setNodeAndParentValidated(ctx); err != nil {
+		return invalidRoots, err
+	}
```

Comment:
We could move this to the callside, but this will be a NOOP anyway since this block will already be valid in essentially every case. We can probably even remove this call at all, if you want.